process_report/process_report.py

```diff
@@ -1,9 +1,108 @@
 import argparse
 import os
+import sys
 
 import pandas
 
 
+### Invoice field names
+INVOICE_MONTH_FIELD = 'Invoice Month'
+PROJECT_FIELD = 'Project - Allocation'
+MANAGER_PI_FIELD = 'Manager (PI)'
+INSTITUTION_FIELD = 'Institution'
+SU_HOUR_FIELD = 'SU Hours (GBhr or SUhr)'
+SU_TYPE_FIELD = 'SU Type'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def apply_credits_0001(dataframe):
+    credit_code = "0001"
+    pass
+
+
+def apply_credits_0002(dataframe):
+    """Applies the New PI Credit. This credit function expects the 
+    env var `C0002_OLD_PI` to be set, pointing to a txt file containing old PIs"""
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except Exception:
+        print("Old PI file does not exist, or the C0002_OLD_PI env var is not set")
+        sys.exit(1)
```

Comment:
Couple of things about this block:
1. Check that "C0002_OLD_PI env var" is set earlier in the code, before trying to open the file. If it's not set, then raise an error and exit.
2. Then, when you open the file, then specifically check for the exception that the file doesn't exist (`FileNotFoundError`). Don't have an all encompassing exception check.

This way you are clear about what caused the failure, instead of saying it could be "X or Y" that caused the failure. For anything else, the exception will be printed.